lib/core/validator.dart

```diff
@@ -35,8 +37,13 @@ class TextValidator extends Validator<String> {
     return value.length > (minLength ?? 0) &&
         (length?.contains(value.length) ?? true) &&
         ((maxLength ?? 0) > 0 ? (value.length <= maxLength!) : true) &&
-        (pattern != null ? match(value) : true);
+        (pattern != null
+            ? (useAdditionalValidation != null &&
+                    useAdditionalValidation!(value)) ||
+                match(value)
+            : true);
```

Comment:
what about the case were `useAdditionalValidation` is not null but returned false?
with the current implementation, it will proceed to check for `match(value)` and if it returned true then it will pass, although it should fail

Reply:
also it would be better and more readable to separate these into variables with meaningful names and then use those variables in the return statement

Reply:
I refactored a bit for better readability in the return of `isValid`, let me know if it looks better, in the `useAdditionalValidation` the idea is just to add on to the RegEx since `3...` addresses are not valid under the `bitcoin_flutter`'s validation function, so if it returns false it fallbacks to the RegEx making sure those addresses can return true